failure to build on Windows (Cygwin)

Using freshly cloned master.  My previous successful build was at 8481fec.
```
HAW7L0605$ make VERBOSE=1
make[1]: Entering directory '/home/s2sajs/julia-master/deps'
mkdir -p /home/s2sajs/julia-master/usr
/usr/bin/tar xzf /home/s2sajs/julia-master/usr-staging/dsfmt-2.2.3.tgz -C /home/s2sajs/julia-master/usr
/usr/bin/tar: ./bin: Cannot open: File exists
/usr/bin/tar: Exiting with failure status due to previous errors
/home/s2sajs/julia-master/deps/dsfmt.mk:43: recipe for target '/home/s2sajs/julia-master/usr/manifest/dsfmt' failed
make[1]: *** [/home/s2sajs/julia-master/usr/manifest/dsfmt] Error 2
make[1]: Leaving directory '/home/s2sajs/julia-master/deps'
Makefile:81: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2
```

--- a/deps/blas.mk
+++ b/deps/blas.mk
@@ -84,7 +84,7 @@ $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/build-compiled: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/
 	echo 1 > $@
 
 define OPENBLAS_INSTALL
-    $(call SHLIBFILE_INSTALL,$1,$2,$3)
+	$(call SHLIBFILE_INSTALL,$1,$2,$3)
 ifeq ($$(OS), Linux)
 	ln -sf libopenblas$$(OPENBLAS_LIBNAMESUFFIX).$$(SHLIB_EXT) $2/$$(build_libdir)/libopenblas$$(OPENBLAS_LIBNAMESUFFIX).$$(SHLIB_EXT).0
 endif

--- a/deps/dsfmt.mk
+++ b/deps/dsfmt.mk
@@ -36,7 +36,8 @@ endif
 	echo 1 > $@
 
 define DSFMT_INSTALL
-	mkdir -p $2/$$(build_includedir) $2/$$(build_libdir)
+	mkdir -p $2/$$(build_includedir)
+	mkdir -p $2/$$(build_shlibdir)
 	cp $1/dSFMT.h $2/$$(build_includedir)
 	cp $1/libdSFMT.$$(SHLIB_EXT) $2/$$(build_shlibdir)
 endef

--- a/deps/llvm.mk
+++ b/deps/llvm.mk
@@ -515,7 +515,7 @@ $(build_prefix)/manifest/llvm: | $(llvm_python_workaround)
 
 ifeq ($(LLVM_USE_CMAKE),1)
 CMAKE_INSTALL_LLVM = \
-	cd $1 && $(CMAKE) -DCMAKE_INSTALL_PREFIX="$2" -P cmake_install.cmake
+	cd $1 && $$(CMAKE) -DCMAKE_INSTALL_PREFIX="$2$$(build_prefix)" -P cmake_install.cmake
 $(eval $(call staged-install,llvm,llvm-$$(LLVM_VER)/build_$$(LLVM_BUILDTYPE), \
 	CMAKE_INSTALL_LLVM,,,))
 else

--- a/deps/mbedtls.mk
+++ b/deps/mbedtls.mk
@@ -73,8 +73,7 @@ $(eval $(call staged-install, \
 
 
 clean-mbedtls:
-	-rm -f $(build_prefix)/manifest/mbedtls \
-		$(BUILDDIR)/mbedtls-$(MBEDTLS_VER)/build-configured \
+	-rm $(BUILDDIR)/mbedtls-$(MBEDTLS_VER)/build-configured \
 		$(BUILDDIR)/mbedtls-$(MBEDTLS_VER)/build-compiled
 	-$(MAKE) -C $(BUILDDIR)/mbedtls-$(MBEDTLS_VER) clean
 

--- a/deps/tools/common.mk
+++ b/deps/tools/common.mk
@@ -129,10 +129,18 @@ endef
 define staged-install
 stage-$(strip $1): $$(build_staging)/$2.tgz
 install-$(strip $1): $$(build_prefix)/manifest/$(strip $1)
-uninstall-$(strip $1): | $$(build_staging)/$2.tgz
+uninstall-$(strip $1):
 	-rm $$(build_prefix)/manifest/$(strip $1)
 	-cd $$(build_prefix) && rm -dv -- $$$$($(TAR) tzf $$(build_staging)/$2.tgz --exclude './$$$$')
+
+ifeq (exists, $$(shell [ -e $$(build_staging)/$2.tgz ] && echo exists ))
+# clean depends on uninstall only if the staged file exists
 distclean-$(strip $1) clean-$(strip $1): uninstall-$(strip $1)
+else
+# uninstall depends on staging only if the staged file doesn't exist
+# otherwise, uninstall doesn't actually want the file to be updated first
+uninstall-$(strip $1): | $$(build_staging)/$2.tgz
+endif
 
 reinstall-$(strip $1):
 	+$$(MAKE) uninstall-$(strip $1)

--- a/deps/tools/git-external.mk
+++ b/deps/tools/git-external.mk
@@ -68,5 +68,5 @@ $5/$$($2_SRC_DIR)/source-extracted: $$($2_SRC_FILE)
 endif # DEPS_GIT
 
 distclean-$1:
-	-rm -rf $5/$$($2_SRC_DIR) $$($2_SRC_FILE) $$(BUILDDIR)/$1 $$(BUILDDIR)/$1-$$($2_SHA1)
+	-rm -rf $5/$$($2_SRC_DIR) $$($2_SRC_FILE) $$(BUILDDIR)/$$($2_SRC_DIR)
 endef

--- a/deps/unwind.mk
+++ b/deps/unwind.mk
@@ -82,7 +82,7 @@ $(build_prefix)/manifest/osxunwind: $(BUILDDIR)/libosxunwind-$(OSXUNWIND_VER)/bu
 	echo $(OSXUNWIND_VER) > $(build_prefix)/manifest/osxunwind
 
 clean-osxunwind:
-	-rm $(build_prefix)/manifest/libuv $(BUILDDIR)/libosxunwind-$(OSXUNWIND_VER)/build-compiled
+	-rm $(build_prefix)/manifest/osxunwind $(BUILDDIR)/libosxunwind-$(OSXUNWIND_VER)/build-compiled
 	-rm -r $(build_libdir)/libosxunwind.a $(build_shlibdir)/libosxunwind.$(SHLIB_EXT) \
 		$(build_includedir)/mach-o/ $(build_includedir)/unwind.h $(build_includedir)/libunwind.h
 	-$(MAKE) -C $(BUILDDIR)/libosxunwind-$(OSXUNWIND_VER) clean $(OSXUNWIND_FLAGS)

